gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -565,7 +592,52 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
 
     return new FSDataInputStream(in);
   }
+  /**Bhagyaa-s**/
+  //Conf not used in the method for sometime, as some changes need to be made for laters
+  private void initThreadPools(Configuration conf){
+    int maxThreads=DEFAULT_MAX_THREADS;
+    //commenting below block as we already hard-coded the value
+//    if (maxThreads < 2) {
+//      LOG.warn("MAX_THREADS or DEFAULT_MAX_THREAD must be at least 2: forcing to 2.");
+//      maxThreads = 2;
+//    }
+    //int totalTasks=32; //commenting as it is used only in select feature of s3
+    int keepAliveTime=60;
+    unboundedThreadPool=new ThreadPoolExecutor(maxThreads,Integer.MAX_VALUE,
+            keepAliveTime,TimeUnit.SECONDS, new LinkedBlockingQueue<>(),
+            BlockingThreadPoolExecutorService.newDaemonThreadFactory("gcs-transfer-unbounded"));
+    unboundedThreadPool.allowCoreThreadTimeOut(true);
+    executorCapacity=16;
+  }
+  @Override
+  public CompletableFuture<FSDataInputStream> openFileWithOptions(final Path rawPath, final OpenFileParameters parameters) throws IOException{
+    final Path path=makeQualified(rawPath);
+    Configuration options=parameters.getOptions();
+    Set<String> mandatoryKeys=parameters.getMandatoryKeys();
+    AbstractFSBuilderImpl.rejectUnknownMandatoryKeys(mandatoryKeys,Collections.emptySet(),"for "+path);
+    StorageResourceId storageResourceId = StorageResourceId.fromUriPath(rawPath.toUri(),false);
+    itemInfo = this.getGcsFs().getGcs().getItemInfo(storageResourceId);
+    LOG.debug("Ignoring file status");
```

Comment:
Please use `logger` for consistency. 

We may want to add file path, and parameters to the logs as well